src/core/src/lib/extensions/field-form/field-form.ts

```diff
@@ -97,21 +97,28 @@ export class FieldFormExtension implements FormlyExtension {
           }
         }
 
-        if (null === c.validator || null === c.asyncValidator) {
+        if (null === c.validator) {
           c.setValidators(() => {
             const v = Validators.compose(this.mergeValidators<ValidatorFn>(field, '_validators'));
 
             return v ? v(c) : null;
           });
-          c.setAsyncValidators(() => {
-            const v = Validators.composeAsync(this.mergeValidators<AsyncValidatorFn>(field, '_asyncValidators'));
-
-            return v ? v(c) : of(null);
-          });
 
           markForCheck = true;
         }
 
+        if (null === c.asyncValidator) {
+          const asyncValidatorsList = this.mergeValidators<AsyncValidatorFn>(field, '_asyncValidators');
```

Comment:
This will break some cases if I'm not wrong, the main reason of calling `mergeValidators` inside `setAsyncValidators` is to keep validation in sync with field changes.

Reply:
If I am understanding correct is that you are saying the the list of asyncValidators for a field can change without the need of rebuilding the field.

Reply:
At runtime I think, when the field is hidden for example the validation should be updated. Try to apply the same changes to `setValidators`, and you'll notice some test failures.

Reply:
I have taken the account of hidden case. And all the test cases are passing.